src/include/stir/scatter/ScatterEstimation.h

```diff
@@ -0,0 +1,343 @@
+    shared_ptr < DiscretisedDensity < 3, float >  > mask_image_sptr;
+
+    //! \details the set of parameters to mask the attenuation image
+    mask_parameters mask_image;
+
+    //! Zoom factor on plane XY. Defaults on 1.f.
+    float zoom_xy;
+    //! Zoom factor on Z axis. Defaults on 1.f.
+    float zoom_z;
+    //! Optional to the zoom_xy the final size of the image can be set
+    int size_xy;
+    //! Optional to the zoom_z the final size of the image can be set
+    int size_z;
+private:
+
+    //! \details A helper function to reduce the size of set_up().ß
+    Succeeded ffw_project_mask_image();
+
+    //! \details A complicated function to mask images.
+    bool apply_mask_in_place(DiscretisedDensity<3, float> &,
+                             const mask_parameters&);
+
+    void add_proj_data(ProjData&, const ProjData&);
+
+    void subtract_proj_data(ProjData&, const ProjData&);
+
+    void apply_to_proj_data(ProjData& , const pow_times_add&);
+
+    //! \details Average the two first activity images 0 and 1.
+    bool do_average_at_2;
+    //! \details Used for convinience. It is initialised on post_processing.
+    bool iterative_method;
+    //!
+    bool remove_interleaving;
+    //! Save all scatter simulated sinograms
+    bool export_scatter_estimates_of_each_iteration;
+    //! Export SSRB sinograms
+    bool export_2d_projdata;
+    //! This bool will allow the ScatterEstimation to override the value of
+    //! the initial activity image set in ScatterSimulation par file
+    bool override_initial_activity_image;
+    //! This bool will allow the ScatterEstimation to override the value of
+    //! the density image set in ScatterSimulation par file
+    bool override_density_image;
+    //! This bool will allow the ScatterEstimation to override the value of
+    //! the density image for scatter points set in ScatterSimulation par file
+    bool override_density_image_for_scatter_points;
+    //! This will over-ride the scanner template in scatter sinogram simulation.
+    bool override_scanner_template;
+    //! In debug mode a lot of extra files are going to be saved in the disk.
+    bool run_debug_mode;
+
+    //! Parameter file for scatter simulation
+    //! \warning Values in this file could be overridden.
+    std::string scatter_sim_par_filename;
+
+    //! \details Class which will implement the scatter simulation.
+    shared_ptr < ScatterSimulation > scatter_simulation_sptr;
+
+    //! This path is used in the debug mode to store all the intermediate files, as they are many.
+    FilePath extras_path;
+
+    shared_ptr<PostFiltering <DiscretisedDensity<3,float> > > filter_sptr;
+
+    //! \details The number of iterations the scatter estimation will perform.
+    //! Default = 5.
+    int num_scatter_iterations;
+
+    //! Default value = 100
+    float max_scale_value;
+    //! Default value = 0.4
+    float min_scale_value;
+    //!
+    unsigned int half_filter_width;
+    //! Ouput file name prefix
```

Comment:
```suggestion
    //! Output file name prefix
```
is this added to the `extras`?